keep/workflowmanager/workflowscheduler.py

```diff
@@ -133,6 +134,67 @@ def _run_workflow(
             )
         self.logger.info(f"Workflow {workflow.workflow_id} ran")
 
+    def handle_workflow_test(self, workflow, tenant_id, triggered_by_user):
+
+        workflow_execution_id = self._get_unique_execution_number()
+
+        self.logger.info(
+            "Adding workflow to run",
+            extra={
+                "workflow_id": workflow.workflow_id,
+                "workflow_execution_id": workflow_execution_id,
+                "tenant_id": tenant_id,
+                "triggered_by": "manual",
+                "triggered_by_user": triggered_by_user,
+            },
+        )
+
+        result_queue = queue.Queue()
+
+        def run_workflow_wrapper(
+            run_workflow, workflow, workflow_execution_id, test_run, result_queue
+        ):
+            try:
+                errors, results = run_workflow(
+                    workflow, workflow_execution_id, test_run
+                )
+                result_queue.put((errors, results))
+            except Exception as e:
+                print(f"Exception in workflow: {e}")
+                result_queue.put((str(e), None))
+
+        thread = threading.Thread(
+            target=run_workflow_wrapper,
+            args=[
+                self.workflow_manager._run_workflow,
+                workflow,
+                workflow_execution_id,
+                True,
+                result_queue,
+            ],
+        )
+        thread.start()
+        thread.join()
+        errors, results = result_queue.get()
```

Comment:
this is my solution for running the workflow sync, maybe there's a cleaner solution? I kept the wrapper inside since I don't believe it will be used anywhere else anytime soon, but the concept could be generalized if we want it to.